fluent-bundle/src/builtins.ts

```diff
@@ -80,13 +80,19 @@ export function NUMBER(
     return new FluentNone(`NUMBER(${arg.valueOf()})`);
   }
 
-  if (arg instanceof FluentNumber || arg instanceof FluentDateTime) {
+  if (arg instanceof FluentNumber) {
     return new FluentNumber(arg.valueOf(), {
       ...arg.opts,
       ...values(opts, NUMBER_ALLOWED)
     });
   }
 
+  if (arg instanceof FluentDateTime) {
+    return new FluentNumber(arg.valueOf(), {
+      ...values(opts, NUMBER_ALLOWED)
+    });
+  }
```

Comment:
`question`: Seems like Gecko doesn't handle that conversion. Do you think it's useful to add or is it something TypeScript specific?

Reply:
TBH, I don't remember if we were intentional about this.

If you pass a datetime to `NUMBER`, you can either do an error or recover by formatting the date as number. I guess in js land, `Number(new Date())` works by design. Also compatible with what `Intl.NumberFormat` does.

Not sure if there are actually good-Fluent arguments for this, though.